### PR TITLE
Migrate PasswordReset.js to typescript

### PR DESCRIPTION
**Pull request by Palid**
---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://611e0d9d70fe6ab1519cfa0c--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/PasswordReset.ts [hunk 1]
```diff
@@ -69,7 +75,7 @@ export default class PasswordReset {
      * with a "message" property which contains a human-readable message detailing why
      * the reset failed, e.g. "There is no mapped matrix user ID for the given email address".
      */
-    async checkEmailLinkClicked() {
+    public async checkEmailLinkClicked() {
```
Comment (SimonBrandner) [inline src/PasswordReset.ts:78]:
```suggestion
    public async checkEmailLinkClicked(): Promise<void> {
```

**Review by SimonBrandner — CHANGES_REQUESTED** [review]
The constructor also has missing types, if you could please add those

File: src/PasswordReset.ts [hunk 2]
```diff
@@ -47,7 +53,7 @@ export default class PasswordReset {
      * @param {string} newPassword The new password for the account.
      * @return {Promise} Resolves when the email has been sent. Then call checkEmailLinkClicked().
      */
-    resetPassword(emailAddress, newPassword) {
+    public resetPassword(emailAddress, newPassword): Promise<IRequestTokenResponse> {
```
Comment (SimonBrandner) [inline src/PasswordReset.ts:56]:
```suggestion
    public resetPassword(emailAddress: string, newPassword: string): Promise<IRequestTokenResponse> {
```

**Review by gsouquet — APPROVED** [review]
What Simon said, otherwise LGTM

**Pull request merged**